src/main/java/com/lambdaworks/redis/SimpleRedisConnection.java

```diff
@@ -0,0 +1,144 @@
+package com.lambdaworks.redis;
+
+import com.lambdaworks.redis.SimpleSerializers.JavaSerializer;
+import com.lambdaworks.redis.codec.ByteArrayCodec;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+/**
+ * @author <a href="mailto:a.abdelfatah@live.com">Ahmed Kamal</a>
+ * @since 12.01.16 09:17
+ */
+
+/**
+ * A simple redis connection for caching and retrieval from redis
+ */
+public class SimpleRedisConnection {
+
+    private final RedisClient redisClient;
+    protected static final InternalLogger logger = InternalLoggerFactory.getInstance(SimpleRedisConnection.class);
+
+    public SimpleRedisConnection(RedisClient redisClient) {
+        this.redisClient = redisClient;
+    }
+
+    /**
+     * Cache generic value in redis (The class of the value should implement Serializable)
+     * <br/>
+     * Sets field in the hash stored at key to value. If key does not exist, a new key holding a hash is created.
+     * <br/>
+     * If field already exists in the hash, it is overwritten.
+     *
+     * @param key
+     * @param field
+     * @param value
+     * @param <T>
+     * @return true if field is a new field in the hash and value was set.
+     * <br/> false if field already exists in the hash and the value was updated.
+     */
+    public <T> boolean cache(String key, String field, T value) {
+
+        RedisConnection<byte[], byte[]> redisCommands = null;
+        try {
+            redisCommands = getRedisConnection();
+
+            byte[] serializedValues = new JavaSerializer().serializeObject(value);
+            redisCommands.hset(key.getBytes(), field.getBytes(), serializedValues);
+
+            return true;
+        } catch (Exception e) {
+            logger.error("Exception Occurred while caching a value in redis", e);
+        } finally {
+            closeConnection(redisCommands);
+        }
+        return false;
+    }
+
+    /**
+     * Get a generic cached value from redis
+     * Returns the value associated with field in the hash stored at key.
+     *
+     * @param key
+     * @param field
+     * @param <T>
+     * @return the value stored in a deserialized form or null when field is not present in the hash or key does not exist.
+     */
+    public <T> T get(String key, String field) {
+
+        RedisConnection<byte[], byte[]> redisCommands = null;
+        try {
+            redisCommands = getRedisConnection();
+
+            byte[] serializedData = redisCommands.hget(key.getBytes(), field.getBytes());
+            T deserializeObject = new JavaSerializer().deserializeObject(serializedData);
+
+            return deserializeObject;
+        } catch (Exception e) {
+            logger.error("Exception Occurred while getting a value from redis", e);
```

Comment:
Catching the exception hides network- and classpath/structural problems and returns `null` to the caller.